applications/sintering/include/pf-applications/sintering/operator_sintering_generic.h

```diff
@@ -270,13 +270,31 @@ namespace Sintering
 
     template <unsigned int with_time_derivative = 2>
     void
-    evaluate_nonlinear_residual(BlockVectorType &      dst,
-                                const BlockVectorType &src) const
+    evaluate_nonlinear_residual(
+      BlockVectorType &      dst,
+      const BlockVectorType &src,
+      const std::function<void(const unsigned int, const unsigned int)>
+        pre_operation = {},
+      const std::function<void(const unsigned int, const unsigned int)>
+        post_operation = {}) const
     {
       MyScope scope(this->timer,
                     "sintering_op::nonlinear_residual",
                     this->do_timing);
 
+      std::function<void(const unsigned int, const unsigned int)>
+        pre_operation_to_be_used = pre_operation;
+
+      if (!pre_operation)
+        pre_operation_to_be_used = [&](const auto start_range,
+                                       const auto end_range) {
+          for (unsigned int b = 0; b < internal::n_blocks(dst); ++b)
+
+            std::memset(internal::block(dst, b).begin() + start_range,
+                        0,
+                        sizeof(Number) * (end_range - start_range));
+        };
+
```

Comment:
It would be also nice to a couple of comments on the default `pre_operation_to_be_used ` is required and what is expected from `pre_operation` in general.